pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Topic.java

```diff
@@ -127,6 +128,14 @@ default long getEntryTimestamp() {
         default void setEntryTimestamp(long entryTimestamp) {
 
         }
+
+        default MessageIdData getMessageIdData() {
+            return null;
+        }
+
+        default ByteBuf getHeaderAndPayload() {
+            return null;
+        }
```

Comment:
It increases two methods to the public interface (`BrokerInterceptor`, which uses `Topic.TopicPublishContext`). We need a PIP for such changes.

For example, https://github.com/apache/pulsar/pull/17269 adds the `getMsgSize`, `getEntryTimestamp`/`setEntryTimestamp` methods to this interface